### DISCUSSION TITLE
Estudar e aplicar novas linguagens no dia a dia.

### DISCUSSION THREAD

**Discussion by ranosi** [frontendbr/forum#801] [Discussão]
Em uma conversa lá no Slack sobre estudos, traçar metas e coisas semelhantes, que começou com esse post bem legal do Glauber http://gabsferreira.com/quer-ficar-bom-em-algo-faca-uma-coisa-de-cada-vez/ , surgiu a questão de: como vocês se organizam e também testam novas linguagens ou frameworks?
Tanto quando surgem coisas novas ou quando estão iniciando nos estudos, de que forma vocês colocam em prática algo novo pra entender como ele funciona e o que tu pode fazer com isso no dia a dia?

**Comment by felipefialho** [comment]
Resumidamente, tento colocar prazos e metas, e testar em projetos open-sources, como escrevi no artigo [Aproximando seus estudos de cenários reais](https://www.felipefialho.com/blog/2015/aproximando-seus-estudos-de-cenarios-reais)

**Comment by lucasbento** [comment]
Minha solução pra testar novas tecnologias é construir projetos que eu ou outras pessoas possam utilizar.

Eu mantenho um tracking de ideias que tenho diariamente (apenas nome e descrição) e sempre que tenho um tempo livre (nem que seja pouco) pego uma delas, implemento em alguma tecnologia que quero aprender/testar e na grande maioria das vezes deixo open-source no meu GitHub.

Outra coisa que faço é **não começar um projeto sem terminar o anterior**, sempre fico num hype quando vejo algo novo e foda e quero parar tudo pra testar, eu cortei essa prática pois não conseguia terminar nenhuma das ideias, a vontade de começar um projeto enquanto ainda está em outro ajuda muito você a terminar o que já tem em mãos.

Eu ouço muito gente dizendo que aprende as coisas só por artigos, livros, vídeos, etc, pra mim isso não funciona muito bem, sempre sinto que pra entender bem algo eu preciso colocar em prática (foi assim que sofri igual um condenado na faculdade) e essa foi a melhor forma pra mim que encontrei até o momento.

**Comment by didiraja** [comment]
Comigo tem dois casos: exercício e necessidade de trabalho.

O exercício é alguma ideia que tive de algo que ou não sei fazer ou é mirabolante pro meu nível. Daí aguardo o momento livre pra executa-lo ou então passo um tempo estudando o assunto pra executar.

O outro é a necessidade específica do trabalho. Meu conhecimento e habilidades está todo direcionado as necessidades lá da agência. E por isso também eu vive mexendo na lista de prioridades. Entoa basicamente, se o job pede ou exige algo novo ou que não sei. Isso vira uma meta pra eu aprende, dominar e depois implementar. Mas algumas dessas demandas não vem diretamente do trabalho mas ajudam muito, como por exemplo o Gulp. Ninguém lá sabe o que é, mas estavam.comecando a exigir resultados melhores de performance.

**Comment by guilherme6191** [comment]
Sigo mais ou menos o estilo do @lucasbento. Porém, antes de começar meus projetinhos tento ler alguns artigos ou fazer cursos de introdução (ex: codecademy.com, udemy, etc) e em seguida parto para o projetinho pessoal. 
Funciona melhor pra mim quando tem prática. Gosto de teoria também, mas só na teoria ou projetos pessoais longos fica cansativo pra mim. De qualquer forma, ao desenvolver o projeto, vou ter que ler docs e etc pra conseguir produzir.

**Comment by fabriciofmsilva** [comment]
Pra galera que curte trabalhar com projetos para aprender, uma coisa bacana pra tentar é ser orientado a um problema. Pense ou procure um problema real, pergunte para a galera, faça uma pesquisa, ou mesmo pense sobre alguma coisa que poderia ser melhorada. Depois parta para fazer um plano e um projeto. Assim você tem uma motivação e um possível produto real.
